source/tutorials/tutorials-dashboard-apps/shell-app.rst

```diff
@@ -0,0 +1,48 @@
+.. _app-development-tutorials-dashboard-apps-shell-app:
+
+Developing the Shell App
+========================
+
+The ``shell`` app is significantly different from the dashboard app. It is purely a ``node.js`` app. 
+This means that we only need to use ``npm`` to build and do dev work.
+
+Create a Dev Shell
+------------------
+#. ``clone`` the OOD repo into: ``cd ~/ondemand/misc`` then ``git clone git@github.com:OSC/ondemand.git``
+#. Work out of the ``dev`` directory/space: ``cd ~/ondemand/dev/``.
+#. Symlink to the ``shell`` app in the cloned repo: ``ln -s ../misc/ondemand/apps/dashboard/shell/ shell``
+#. Make a branch and to begin word on your dev dashboard: ``git checkout -b dev_work`` 
+#. Rebuild the ``shell`` app:  ``bin/setup``.
+#. Navigate to the Sandbox and launch your dev ``shell`` app to use the ``dev_work`` branch.
+
+Notice the ``url`` for this app. No longer do you  see ``*/sys/shell`` but instead ``*/dev/shell``. 
+
+Build Old Version of Shell
+..........................
+
+#. pull down ondemand in ``~/ondemand/misc``
+#. then list versions with ``git tags``
+#. run ``git checkout <version>`` 
+#. now make a branch with your changes and use ``bin/setup`` to rebuild if needed
+
+Then for the **shell app** from a login terminal run:
+
+#. ``cd dev``
+#. ``ln -s ../misc/ondemand/apps/dashboard/shell/ shell-18``
+
+And then step into that symlinked directory and run:
+#. ``bin/setup`` 
+
+If you go start this app from sandbox you'll notice a different url than before, 
+showing you are in a dev shell: ``pun/dev/shell-1.8``
+
+Issues and Errors
+.................
+**If above fails** try to purge the modules with ``module purge`` on the login node first, 
+then ensure the *correct ruby version* with ``module load ruby/3.0.2`` then rebuild with ``bin/setup`` as above.
```

Comment:
Is this in the right place? It's talking about `ruby` module but it's in the shell's tutorial?

I do however like the idea of having an issues section that can help either/both tutorials.

Reply:
Also note that the ruby version is subject to change (it's 3.1 in OOD 3.1) but also dependent on the environment.  We have `ruby/3.0.2` but any ruby 3.0 version would work.

Though again, it turns into 3.1 in OOD 3.1 so we may need to use/create a jinja variable for the ruby version (there may already be one).

Reply:
I think that is a remnant of the old notes I built this from, which itself was copied largely from the dashboard notes, so that's how that snuck in I think. I can remove that.